core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -136,6 +136,16 @@ $(document).ready(function() {
             (attempt.hasOwnProperty("attemptId") ? attempt["attemptId"] + "/" : "") + "logs";
           attempt["durationMillisec"] = attempt["duration"];
           attempt["duration"] = formatDuration(attempt["duration"]);
+          var idStr = id.toString();
+          if(idStr.indexOf("application_") > -1) {
+            attempt["master"] = "yarn";
+          } else if(idStr.indexOf("app-") > -1) {
+            attempt["master"] = "standalone";
+          } else if(idStr.indexOf("local-") > -1) {
+            attempt["master"] = "local";
+          } else {
+            attempt["master"] = "mesos";
```

Comment:
Separate from my other comment this isn't a good way to determine the resource manager. Users can set the application name on application start to whatever they want. These are just the defaults for when the parameter isn't set and they've actually changed themselves from one version of Spark to the next previously.

Reply:
I am through the ID to judge, not through the name to judge. ID is based on the type of resource management to determine.

Reply:
Ok so I was mixing up `appId` and `name` but the issue still applies, this is not a very good way to determine the RM, plus you missed the default case that is used in tests and in case of error (`"spark-application-"`). Overall I'm still not a fan of this PR though